beginner_source/maskedtensor_overview.rst

```diff
@@ -0,0 +1,244 @@
+++++++++++++++++++
+
+Construction
+------------
+
+There are a few different ways to construct a MaskedTensor:
+
+* The first way is to directly invoke the MaskedTensor class
+* The second (and our recommended way) is to use :func:`masked.masked_tensor` and :func:`masked.as_masked_tensor` factory functions,
+  which are analogous to :func:`torch.tensor` and :func:`torch.as_tensor`
+
+  .. autosummary::
+    :toctree: generated
+    :nosignatures:
+
+    masked.masked_tensor
+    masked.as_masked_tensor
+
+Accessing the data and mask
+---------------------------
+
+The underlying fields in a MaskedTensor can be accessed through:
+
+* the :meth:`MaskedTensor.get_data` function
+* the :meth:`MaskedTensor.get_mask` function. Recall that ``True`` indicates "specified" or "valid" while ``False`` indicates
+  "unspecified" or "invalid".
+
+In general, the underlying data that is returned may not be valid in the unspecified entries, so we recommend that
+when users require a Tensor without any masked entries, that they use :meth:`MaskedTensor.to_tensor` (as shown above) to
+return a Tensor with filled values.
+
+Indexing and slicing
+--------------------
+
+:class:`MaskedTensor` is a Tensor subclass, which means that it inherits the same semantics for indexing and slicing
+as :class:`torch.Tensor`. Below are some examples of common indexing and slicing patterns:
+
+    >>> data = torch.arange(60).reshape(3, 4, 5)
+    >>> mask = data % 2 == 0
+    >>> mt = masked_tensor(data.float(), mask)
+    >>> mt[0]
+    MaskedTensor(
+      [
+        [  0.0000,       --,   2.0000,       --,   4.0000],
+        [      --,   6.0000,       --,   8.0000,       --],
+        [ 10.0000,       --,  12.0000,       --,  14.0000],
+        [      --,  16.0000,       --,  18.0000,       --]
+      ]
+    )
+    >>> mt[[0,2]]
+    MaskedTensor(
+      [
+        [
+          [  0.0000,       --,   2.0000,       --,   4.0000],
+          [      --,   6.0000,       --,   8.0000,       --],
+          [ 10.0000,       --,  12.0000,       --,  14.0000],
+          [      --,  16.0000,       --,  18.0000,       --]
+        ],
+        [
+          [ 40.0000,       --,  42.0000,       --,  44.0000],
+          [      --,  46.0000,       --,  48.0000,       --],
+          [ 50.0000,       --,  52.0000,       --,  54.0000],
+          [      --,  56.0000,       --,  58.0000,       --]
+        ]
+      ]
+    )
+    >>> mt[:, :2]
+    MaskedTensor(
+      [
+        [
+          [  0.0000,       --,   2.0000,       --,   4.0000],
+          [      --,   6.0000,       --,   8.0000,       --]
+        ],
+        [
+          [ 20.0000,       --,  22.0000,       --,  24.0000],
+          [      --,  26.0000,       --,  28.0000,       --]
+        ],
+        [
+          [ 40.0000,       --,  42.0000,       --,  44.0000],
+          [      --,  46.0000,       --,  48.0000,       --]
+        ]
+      ]
+    )
+
+Semantics
++++++++++
+
+MaskedTensor vs NumPy's MaskedArray
```

Comment:
I feel like this can be moved to the end? The flow of the tutorial should be:

what is a MT
How do you access/slice
What operations can you do / reductions
How does this compare with alternatives to masked tensor

Reply:
Discussed offline - this will include "what is a MT and why is an MT useful"